src/components/SpruceForm/Widgets/utils.ts

```diff
@@ -0,0 +1,38 @@
+import { Errors } from "../errors";
+
+/**
+ * Returns true if a given value is null or undefined.
+ */
+export const isNullish = (val: any) => val === null || val === undefined;
+
+/**
+ * Processes errors by removing "invisible" and duplicate errors.
+ */
+export const processErrors = (errors: string[]) =>
```

Comment:
[nit] Since we always use the result of this function to calculate `hasErrors` and an error string, we could directly return an object with those fields